dart/src/main/java/com/f2prateek/dart/internal/IntentBuilder.java

```diff
@@ -0,0 +1,86 @@
+package com.f2prateek.dart.internal;
+
+import java.util.Iterator;
+import java.util.Map;
+
+final class IntentBuilder {
+  private final Map<String, ExtraInjection> injectionMap;
+  private final String classPackage;
+  private final String className;
+  private final String targetClass;
+
+  IntentBuilder(String classPackage, String className, String targetClass, Map<String, ExtraInjection> injectionMap) {
+    this.classPackage = classPackage;
+    this.className = className;
+    this.targetClass = targetClass;
+    this.injectionMap = injectionMap;
+  }
+
+  String getFqcn() {
+    return classPackage + "." + className;
+  }
+
+  String brewJava() {
+    StringBuilder builder = new StringBuilder();
+    builder.append("// Generated code from Dart. Do not modify!\n");
+    builder.append("package ").append(classPackage).append(";\n\n");
+    builder.append("import android.content.Context;\n");
+    builder.append("import android.content.Intent;\n\n");
+
+    builder.append("public class ").append(className).append(" {\n");
+    builder.append("  private final Context context;\n\n");
+
+    createConstructor(builder);
+
+    for (ExtraInjection injection : injectionMap.values()) {
+      createSetter(injection, builder);
+    }
+
+    createBuildMethod(builder);
+
+    builder.append("}\n");
+    return builder.toString();
+  }
+
+  private void createSetter(ExtraInjection injection, StringBuilder builder) {
```

Comment:
You should use a templating lib here, this is super hard to read and maintain : 
* use velocity, a well know templating library, pretty easy to use.
* or use JavaPoet from square.